RFCs/0018-Skip-Default-Adapters.md

```diff
@@ -0,0 +1,49 @@
+# 0018 Skip Default Adapters
+
+## Summary
+This note outlines the proposed changes for adding option for skipping initialization of default adapters and thus skipping default adapters to take part in test discovery/execution.
+
+## Motivation
+In addition to test adapters provided via TestAdaptersPaths, test platform loads default adapters present in Extensions folder. This folder is present alongside vstest.console.exe. Even for test sources which can't be discovered/executed by default adapters, we pass those test sources to default adapters. This causes following problems:
+1. Adapters initilization cost increases.
+2. Discovery/Execution time taken by adapters increases as test source is passed to default adapters as well for discovery/execution.
+
+Example: Default adapters can't run/discover XUnit tests, still testplatform initializes default adapters and pass XUnit test assemblies to them to discover/execute.
+
+Proposed changes are focused on improving performance of test discovery/execution by allowing option to skip initialization of default adapters.
+
+## Proposed Changes
+1. Design mode clients can skip default adapters by setting TestPlatformOptions.SkipDefaultAdapters as true.
```

Comment:
why specific to design mode client

Reply:
We are enabling this scenario for design mode only for now.
As tester is not aware of default extensions being used, not adding it as command line options for now. Will add the same if same demand comes for command line.

Reply:
does this mean folks will write design mode client specific to test sources. how will they know they don't to initialize the default adapters